guides/source/configuring.md

```diff
@@ -3231,6 +3231,13 @@ configured without modifying the URL set in the deployment environment. See:
 
 TIP: You don't have to update the database configurations manually. If you look at the options of the application generator, you will see that one of the options is named `--database`. This option allows you to choose an adapter from a list of the most used relational databases. You can even run the generator repeatedly: `cd .. && rails new blog --database=mysql`. When you confirm the overwriting of the `config/database.yml` file, your application will be configured for MySQL instead of SQLite. Detailed examples of the common database connections are below.
 
+By default, Rails will use the standard CLI interface for your database when running `rails dbconsole` (e.g. `psql` for PostgreSQL and `mysql` for MySQL). If you wish to use an alternative application, you can specify this with the `cmd` key, e.g.
+
+```yaml
+development:
+  cmd: pgcli
```

Comment:
I don't like `cmd`. It's not clear enough. How about `database_cli`? I'm also not sure this should be per-database. Is there any use case where one db would want to use `pgsql` and another `pgcli`?

Reply:
Agreed on the first point. I thought it might be common to want to use something like `pgcli` locally but not want to install that on your production servers. But of course it’s easy enough to just define it in `default:` and optionally override, unless I’m misunderstanding and you mean it should be defined somewhere else entirely?

Reply:
I don't mean to define it in `default`, I mean maybe it should be an attr_accessor on ActiveRecord like `schema_cache_ignored_tables`. IE `ActiveRecord.database_cli = :pgcli` or something like that. It could even support a hash of values if we considered you might be using mysql and postgres in a single app (`ActiveRecord.database_cli = { postgresql: :pgcli, mysql: :mysql_cli }`

I think this example would be pretty super rare and error prone, I'm not sure it should be supported via the database config:

```yaml
development:
  primary:
    database: "my_db"
    adapter: "postgres"
    database_cli: "pgsql"
  secondary:
    database: "my_other_db"
    adapter: "postgres"
    database_cli: "pgcli"
```

Reply:
Okay, I think that makes sense. Just to be sure I understand before I take a shot at this tomorrow, you’re suggesting:
- Add a `database_cli` attr_accessor to ActiveRecord
- This would define the default CLI for each database adapter
- Users would override this with `config.active_record.database_cli = { postgres: "pgcli" }`

Reply:
I’ve changed the PR to work in that way, and I think the result is much nicer thanks. I’ve also squashed the commits ready to merge if you’re happy with it at this point.

Is there anywhere else this should be documented to encourage other database adapter authors to use the convention?

There is a test failure currently but I think it’s something intermittent as I’m not getting it locally and it only occurred after squashing, not any actual code change.